src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -108,14 +112,21 @@ public function createView(ChoiceListInterface $list, $preferredChoices = null,
                 $attr,
                 $preferredChoices,
                 $preferredViews,
+                $preferredViewsOrder,
                 $otherViews
             );
         }
 
+        uksort($preferredViews, static function ($a, $b) use ($preferredViewsOrder): int {
+            return isset($preferredViewsOrder[$a], $preferredViewsOrder[$b])
```

Comment:
This check is needed when we have groups in preferred views. 
This happens in tests and might happen in user land.